### ISSUE TITLE
Enable builtin project.scmManager/dependencies/team for maven-plugin 

### ISSUE THREAD

**Issue by GregDomjan** [bsorrentino/maven-confluence-plugin#100]
Proposing for enhancement
I'd like to add scmManager/dependencies/team to a maven-plugin similar to a 'regular' maven project.

It's not currently visible in the documentation for built-in template variables, but there is different handling for them based on being a maven-plugin project or not.

I found looking at the source for this plugin the `${plugin.*` variables so that goals can be listed.

I'm trying this out for first time on maven-plugin and finding that the documented template `${project.*` variables are not available, on further testing the `${gitlog.*` variables are also not available.

Had a quick trawl through the code pased on recent #93 checkin.
Seems that depending on project type being manen-plugin or not it's firing
`generatePluginReport` or `generateProjectReport`
`${project.*` variables only seem to be present in `generateProjectReport`

**Comment by bsorrentino** [comment]
Hi 

The **```${project.*``` variables should be available** on plugin documentation

Let me check

**Comment by GregDomjan** [comment]
In case it helps for reference

Using any of these is failing with similar to the below exception
```
h1. Project Documentation

${gitlog.jiraIssues}
${project.team}
${project.description}
${project.scmManager}
${project.dependencies}
```


```
[INFO] Confluence version [5.6.3-5636] development version [false]
[INFO] speceKey=A parentPageTitle=A
[ERROR] has been imposssible connect to confluence due exception
org.apache.maven.tools.plugin.generator.GeneratorException: java.lang.Exception: com.atlassian.confluence.rpc.RemoteException: com.atlassian
.confluence.content.render.xhtml.migration.exceptions.UnknownMacroMigrationException: The macro 'gitlog.jiraissues' is unknown.
        at org.bsc.maven.confluence.plugin.PluginConfluenceDocGenerator.execute(PluginConfluenceDocGenerator.java:237)
        at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo$2.execute(ConfluenceDeployMojo.java:668)
        at org.bsc.maven.confluence.plugin.AbstractBaseConfluenceMojo.confluenceExecute(AbstractBaseConfluenceMojo.java:189)
        at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.generatePluginReport(ConfluenceDeployMojo.java:648)
        at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.execute(ConfluenceDeployMojo.java:254)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.codehaus.swizzle.confluence.ConfluenceException: java.lang.Exception: com.atlassian.confluence.rpc.RemoteException: com.atlas
sian.confluence.content.render.xhtml.migration.exceptions.UnknownMacroMigrationException: The macro 'gitlog.jiraissues' is unknown.
        at org.codehaus.swizzle.confluence.Confluence.call(Confluence.java:971)
        at org.codehaus.swizzle.confluence.Confluence.storePage(Confluence.java:359)
        at org.bsc.maven.confluence.plugin.PluginConfluenceDocGenerator.processMojoDescriptors(PluginConfluenceDocGenerator.java:376)
        at org.bsc.maven.confluence.plugin.PluginConfluenceDocGenerator.execute(PluginConfluenceDocGenerator.java:234)
        ... 26 more
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------

```

**Comment by bsorrentino** [comment]
Hi i've worked on issue. I have release a ```4.7-SNAPSHOT``` with fix

Please,  could you test it and let me know ?

Thanks in advance

**Comment by GregDomjan** [comment]
The variables ```${project.team} ${project.scmManager} ${project.dependencies}``` are now available, but had a bit of confusion on summary/description.
Not sure if I'm just not configured for the jiraissues to be populated.


I got the following, thinking you had maybe removed plugin.summary to just provide project.description, I changed the confluence to project.description and it also failed.
I then noticed the warning and tried project.summary and it allowed that.

 The macro 'plugin.summary' is unknown.
 The macro 'project.description' is unknown.
 The macro 'gitlog.jiraissues' is unknown.

```
[INFO] --- confluence-reporting-maven-plugin:4.8-SNAPSHOT:deploy (default-cli) @ websign-maven-plugin ---
...
[WARNING] variable plugin.summary or project.summary not defined in template
[ERROR] has been imposssible connect to confluence due exception
org.codehaus.swizzle.confluence.ConfluenceException: java.lang.Exception: com.atlassian.confluence.rpc.RemoteException: com.atlassian.confluence.content.render.xhtml.migration.exceptions.UnknownMacroMigrationException: The macro 'plugin.summary' is unknown.
        at org.codehaus.swizzle.confluence.Confluence.call(Confluence.java:971)
        at org.codehaus.swizzle.confluence.Confluence.storePage(Confluence.java:359)
        at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo$PluginGenerator.processMojoDescriptors(ConfluenceDeployMojo.java:862)
        at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo$2.execute(ConfluenceDeployMojo.java:692)
        at org.bsc.maven.confluence.plugin.AbstractBaseConfluenceMojo.confluenceExecute(AbstractBaseConfluenceMojo.java:189)
        at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.generatePluginReport(ConfluenceDeployMojo.java:676)
        at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.execute(ConfluenceDeployMojo.java:264)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
```

I commented (adding extr {) variables in the template that failed trying different things until the following loaded.

```
h1. Index
{panel}
{toc}
{panel}

h1. Introduction

${{plugin.summary}
${{project.description}
${project.summary}

h1. Usage

${plugin.goals}

${artifactId}:${version}

${pageTitle}


h1. Project Documentation

${{gitlog.jiraIssues}

${project.team}

${{project.description}


${project.scmManager}


${project.dependencies}
```

**Comment by bsorrentino** [comment]
thanks for feedback 

I'll check soon

**Comment by bsorrentino** [comment]
Actually the built-in variables support for **plugin document generation** are:  

 variable | supported | note
---|---|---
```${plugin.summary}``` |  yes |  available only in home page. **deprecated**, use ```${project.summary}``` instead
```${plugin.goals}```| yes | available only in home page 
```${project.scmManager}``` | yes 
```${{project.summary}```| yes
```${project.team}```|yes
```${project.dependencies}```|yes
```${artifactId}```| yes 
```${version}``` | yes
```${pageTitle}``` | yes 
```${gitlog.jiraIssues}``` | yes | available only in home page  and if [gitLogJiraIssuesEnable](http://bsorrentino.github.io/maven-confluence-plugin/deploy-mojo.html#gitLogJiraIssuesEnable) is **true**
${project.description}| **no** 

I have releases a new ```4.8-SNAPSHOT```

**Comment by GregDomjan** [comment]
Looking good to me.

**Comment by bsorrentino** [comment]
fix released on 4.8
